webui/react/src/components/UserSettingsModal.tsx

```diff
@@ -0,0 +1,84 @@
+import { useObservable } from 'micro-observables';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { Modal } from 'components/kit/Modal';
+import userSettings from 'stores/userSettings';
+import { isObject } from 'utils/data';
+import handleError from 'utils/error';
+import { Loadable, Loaded } from 'utils/loadable';
+
+import CodeEditor from './kit/CodeEditor';
+
+interface Props {
+  onSave?: () => void;
+}
+
+const UserSettingsModal: React.FC<Props> = ({ onSave }: Props) => {
+  const loadableState = useObservable(userSettings.getAll());
+  const stringifiedState = Loadable.map(loadableState, (state) =>
```

Comment:
Since you're only ever using `loadableState` here, you'd be better off doing this inside a `select` on the observable so that you maintain object identity.